scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+        returns ``nan``.
+
+        See Also
+        --------
+        ContinuousDistribution.icdf
+        ContinuousDistribution.ilogccdf
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> import numpy as np
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-0.5, b=0.5)
+
+        Evaluate the inverse CCDF at the desired argument:
+
+        >>> X.iccdf(0.25)
+        0.25
+        >>> np.allclose(X.iccdf(0.25), X.icdf(1-0.25))
+        True
+
+        This function returns NaN when the argument is outside the domain.
+
+        >>> X.iccdf([-0.1, 0, 1, 1.1])
+        array([ nan,  0.5, -0.5,  nan])
+
+        """
+        return self._iccdf_dispatch(x, method=method, **self._parameters)
+
+    @_dispatch
+    def _iccdf_dispatch(self, x, method=None, **kwargs):
+        if self._overrides('_iccdf_formula'):
+            method = self._iccdf_formula
+        elif self.tol is _null and self._overrides('_icdf_formula'):
+            method = self._iccdf_complementarity
+        else:
+            method = self._iccdf_inversion
+        return method
+
+    def _iccdf_formula(self, x, **kwargs):
+        return NotImplementedError(self._not_implemented)
+
+    def _iccdf_complementarity(self, x, **kwargs):
+        return self._icdf_dispatch(1 - x, **kwargs)
+
+    def _iccdf_inversion(self, x, **kwargs):
+        return self._solve_bounded(self._ccdf_dispatch, x, kwargs=kwargs)
+
+    ### Sampling Functions
+    # The following functions for drawing samples from the distribution are
+    # exposed via a public method that accepts one positional argument - the
+    # shape of the sample - and keyword options (but not distribution
+    # parameters).
+    # sample
+    # ~~qmc_sample~~ built into sample now
+    #
+    # Common keyword options include:
+    # method - a string that indicates which method should be used to compute
+    #          the quantity (e.g. a formula or numerical integration).
+    # rng - the NumPy Generator object to used for drawing random numbers.
+    #
+    # Input/output validation is included in each function, since there is
+    # little code to be shared.
+    # These are the methods meant to be called by users.
+    #
+    # Each public method calls a private "dispatch" method that
+    # determines which "method" (strategy for calculating the desired quantity)
+    # to use by default and, via the `@_dispatch` decorator, calls the
+    # method and computes the result.
+    # Each dispatch method can designate the responsibility of sampling to any
+    # of several "implementation" methods. These methods accept only
+    # `**kwargs`, the parameter dictionary passed from the public method via
+    # the "dispatch" method.
+    # See the note corresponding with the "Distribution Parameters" for more
+    # information.
+
+    def sample(self, shape=(), *, method=None, rng=None, qmc_engine=None):
```

Comment:
Add a way to change the dimensionality of a `qmc_engine` instance, and we can do that.

Reply:
I am not sure I understand the link. We can discuss this IRL tomorrow.

Reply:
We concluded that if the user passes in a 1D QMCEngine, it could be accepted as `rng`, and we would do the same thing as is done [under the hood in `qmc_quad`](https://github.com/scipy/scipy/blob/7dcd8c59933524986923cde8e9126f5fc2e6b30b/scipy/integrate/_quadrature.py#L1813-L1825), generating a QMCEngine with a different `rng` for each axis-slice.